docs/SPIR-V.rst

```diff
@@ -1923,6 +1940,65 @@ Example 3: (compiled with ``-fvk-bind-globals 2 1``)
 Note that if the developer chooses to use this command line option, it is their
 responsibility to provide proper numbers and avoid binding overlaps.
 
+ResourceDescriptorHeaps & SamplerDescriptorHeaps
+------------------------------------------------
+
+The SPIR-V backend supported SM6.6 resource heaps, using 2 extensions:
+- `SPV_EXT_descriptor_indexing`
+- `VK_EXT_mutable_descriptor_type`
+
+The descriptor 0 is used to address resources in both the sampler and resource
+heaps. The binding index is the index used to subscript in the heap.
+
+.. code:: hlsl
+   Texture2D tex = ResourceDescriptorHeap[10];
+   // tex is in the descriptor set 0, binding 10.
```

Comment:
Vulkan binding is different than DX. In Vulkan, an array of resources do not take up a binding for each element of the array. As long as the option `-fspv-flatten-resource-arrays` is not used, the entire descriptor heap should only use a single binding in Vulkan. I don't know if that changes your ideas on how to make this work with implicit bindings.